python/CLOUDFRONT_VIEWER_POLICY_HTTPS/CLOUDFRONT_VIEWER_POLICY_HTTPS_test.py

```diff
@@ -0,0 +1,210 @@
+import json
+import sys
+import unittest
+try:
+    from unittest.mock import MagicMock
+except ImportError:
+    from mock import MagicMock
+import botocore
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::CloudFront::Distribution'
+
+#############
+# Main Code #
+#############
+
+CONFIG_CLIENT_MOCK = MagicMock()
+STS_CLIENT_MOCK = MagicMock()
+
+class Boto3Mock():
+    @staticmethod
+    def client(client_name, *args, **kwargs):
+        if client_name == 'config':
+            return CONFIG_CLIENT_MOCK
+        if client_name == 'sts':
+            return STS_CLIENT_MOCK
+        raise Exception("Attempting to create an unknown client")
+
+sys.modules['boto3'] = Boto3Mock()
+
+RULE = __import__('CLOUDFRONT_VIEWER_POLICY_HTTPS')
+
+class SampleTest(unittest.TestCase):
+
+    invoking_event_iam_role_sample = '{"configurationItem":{"relatedEvents":[],"relationships":[],"configuration":{},"tags":{},"configurationItemCaptureTime":"2018-07-02T03:37:52.418Z","awsAccountId":"123456789012","configurationItemStatus":"ResourceDiscovered","resourceType":"AWS::IAM::Role","resourceId":"some-resource-id","resourceName":"some-resource-name","ARN":"some-arn"},"notificationCreationTime":"2018-07-02T23:05:34.445Z","messageType":"ConfigurationItemChangeNotification"}'
+
+    resource_id = ['E0123456789012', 'E1234567891234', 'E234567890123']
+
+    configurations = [{'distributionConfig':{'cacheBehaviors':{'quantity':2, 'items':[{'viewerProtocolPolicy':'allow-all', 'pathPattern':'images/*.jpg'}, {'viewerProtocolPolicy':'allow-all', 'pathPattern':'videos/*.mp4'}]}, 'defaultCacheBehavior': {'viewerProtocolPolicy':'redirect-to-https'}}}, {'distributionConfig':{'cacheBehaviors':{'quantity':0}, 'defaultCacheBehavior':{'viewerProtocolPolicy':'allow-all'}}}, {'distributionConfig':{'cacheBehaviors':{'quantity':0}, 'defaultCacheBehavior':{'viewerProtocolPolicy':'redirect-to-https'}}}]
+
+    #Gerkin Scenario 1a: Default viewer protocol policy is set to 'allow-all'
+    def test_scenario1a(self):
+        invoking_event = construct_invoking_event(construct_config_item(self.configurations[1], self.resource_id[1]))
+        lambda_event = build_lambda_configurationchange_event(invoking_event, rule_parameters=None)
+        response = RULE.lambda_handler(lambda_event, {})
+        resp_expected = []
+        resp_expected.append(build_expected_response('NON_COMPLIANT', 'E1234567891234', 'AWS::CloudFront::Distribution', annotation='''Default ViewerProtocolPolicy is set to 'allow-all' for this Amazon CloudFront distribution.'''))
+        assert_successful_evaluation(self, response, resp_expected)
+
+    #Gerkin Scenario 1b: COMPLIANT
+    def test_scenario1b(self):
+        invoking_event = construct_invoking_event(construct_config_item(self.configurations[2], self.resource_id[2]))
```

Comment:
combine both functions into one construct_invoking_event() and move them above "#helper functions" comment.